tests/test_report.py

```diff
@@ -0,0 +1,543 @@
+                        "path": "/nonexistent",
+                        "size": 384,
+                    },
+                    {
+                        "__typename__": "FileMagicReport",
+                        "magic": "Zip archive data, at least v2.0 to extract",
+                        "mime_type": "application/zip",
+                    },
+                    {
+                        "__typename__": "ChunkReport",
+                        "end_offset": 384,
+                        "extraction_reports": [],
+                        "handler_name": "zip",
+                        "id": "test_basic_conversion:id",
+                        "is_encrypted": False,
+                        "size": 384,
+                        "start_offset": 0,
+                    },
+                ],
+                "subtasks": [
+                    {
+                        "__typename__": "Task",
+                        "chunk_id": "test_basic_conversion:id",
+                        "depth": 314,
+                        "path": "/extractions/nonexistent_extract",
+                    }
+                ],
+                "task": {
+                    "__typename__": "Task",
+                    "chunk_id": "",
+                    "depth": 0,
+                    "path": "/nonexistent",
+                },
+            },
+        ]
+
+    def test_exotic_command_output(self):
+        task = Task(path=Path("/nonexistent"), depth=0, chunk_id="")
+        task_result = TaskResult(task)
+        report = ExtractCommandFailedReport(
+            command="dump all bytes",
+            stdout=bytes(range(256)),
+            stderr=b"stdout is pretty strange ;)",
+            exit_code=1,
+        )
+
+        task_result.add_report(
+            ChunkReport(
+                id="test",
+                handler_name="fail",
+                start_offset=0,
+                end_offset=256,
+                size=256,
+                is_encrypted=False,
+                extraction_reports=[report],
+            )
+        )
+        json_text = ProcessResult(results=[task_result]).to_json()
+
+        decoded_report = json.loads(json_text)
+
+        assert decoded_report == [
+            {
+                "__typename__": "TaskResult",
+                "reports": [
+                    {
+                        "__typename__": "ChunkReport",
+                        "end_offset": 256,
+                        "extraction_reports": [
+                            {
+                                "__typename__": "ExtractCommandFailedReport",
+                                "command": "dump all bytes",
+                                "exit_code": 1,
+                                "severity": "WARNING",
+                                "stderr": "stdout is pretty strange ;)",
+                                "stdout": (
+                                    "b'\\x00\\x01\\x02\\x03\\x04\\x05\\x06\\x07"
```

Comment:
This looks wrong, the binary contents seems to be double-encoded here.

Reply:
This is a regression test, that was triggered by a failing command.

Although the output looks non-trivial/funny, there is no double-encoding here - it is the python `repr` of a byte-string.

-----

The original content (stdout) is a byte-array, which means that it needs to be converted somehow to a readable representation for JSON, whenever possible it is decoded as UTF-8 and the string representation is what we expect, however in this case we really have a binary output, so we default to the Python `repr`.